sk0-front/src/app/common/services/registerEmailValidator.ts

```diff
@@ -0,0 +1,26 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from "@angular/forms";
+import { Observable, of } from "rxjs";
+
+export function RegisterEmailValidator(): ValidatorFn {
+  return (control: AbstractControl): ValidationErrors | null => {
+    if (!control.value) {
+      return null;
+    }
+
+    const [username, serverDomain] = control.value.split("@");
+
+    if (!username || !serverDomain) {
+      return null;
+    }
+
+    const [server, domain] = serverDomain.split(".");
```

Comment:
```suggestion
    const [, domain] = serverDomain.split(".");
```
if the first element is not used, you can omit it like this or prefix with an `_`, by convention.

```suggestion
    const [_server, domain] = serverDomain.split(".");
```